app/controllers/JourneyAnswersController.scala

```diff
@@ -79,11 +81,13 @@ class JourneyAnswersController @Inject()(propertyService: PropertyService,
       withJourneyContextAndEntity[Expenses](taxYear, incomeSourceId, nino, JourneyName.RentalExpenses, request) { (_, expenses) =>
         handleResponse(CREATED) {
           for {
+            psr <- propertyService.getCurrentPeriodicSubmission(taxYear.endYear, nino.value, nino.value, incomeSourceId.value) //.getPropertyPeriodicSubmissions(taxYear, incomeSourceId)
```

Comment:
Move this logic to service layer and split the responsibility

Reply:
Done.